velox/docs/develop/types.rst

```diff
@@ -73,6 +73,27 @@ Examples:
 * Timestamp(-5000*24*60*60 - 1000, 123456) represents 1956-04-24 07:43:20.000123456
   (5000 days 1000 seconds before epoch plus 123456 nanoseconds).
 
+Floating point types (REAL, DOUBLE) have special values negative/positive infinity and
+not-a-number (NaN).
+
+For NaN the semantics are different than the standard floating point semantics:
+
+* NaN = NaN returns true.
+* NaN is treated as a normal value in join and group-by keys.
+* NaN values go last when in ascending order, larger than any other numeric value.
+
+For negative/positive infinity the following standard floating point semantics apply:
+
+* Positive infinity multiplied by any positive value returns positive infinity.
```

Comment:
I wonder if it helps to add a shorter version of these rules as well.

Given, V a positive finite number,

```
* +inf * V = +inf (Positive infinity multiplied by any positive value returns positive infinity.)
* -inf * V = -inf (Positive infinity multiplied by any negative value returns negative infinity.)
* ...
```